SoftwareGuide/Latex/DevelopmentGuidelines/CreateAModule.tex

```diff
@@ -929,6 +930,151 @@ \subsubsection{Wrapping Tests}
+    \code{CMAKE\_BUILD\_TYPE:STRING="RelWithDebInfo"} and \code{ITK\_WRAP\_PYTHON:BOOL="On"}.
+    Note that the "RelWithDebInfo" build type is strongly encouraged over a "Debug"
+    build as the former will build against a standard Python distribution.
+    See Section~\ref{sec:UsingCMakeForConfiguringAndBuildingITK} for a detailed
+    explanation of how to build ITK locally with CMake.
+  \item A Python virtual environment must be appropriately configured with
+    \code{WrapITK.pth} so that the ITK debug build can be loaded.
+    See Section~\ref{sec:Wrapping} for further explanation.
+  \item ITK modules to be debugged must be loaded in a new Python session initialized
+    from the given virtual environment. Given that ITK Python uses lazy loading, it is
+    pragmatic to use \code{itk.force\_load()} to ensure that all possible debug symbols
+    are made available. The \code{os} module can be used to identify the PID of the given session.
+
+\begin{minted}[baselinestretch=1,fontsize=\footnotesize,linenos=false,bgcolor=ltgray]{bash}
+(venv-itk) > python
+>>> import itk
+>>> itk.force_load()
+>>> import os
+>>> os.getpid()
+99999
+\end{minted}
+
+  \item The Python session may be attached to a debugger using the returned PID.
+    \begin{itemize}
+      \item On a Windows operating system Microsoft Visual Studio 2019 or a similar platform
+        can be used for attaching to a running process for debugging. Select "Attach To Process"
+        from the Debug menu, choose "Native" code, and then search for the process PID. If
+        debug symbols were generated correctly then ITK modules will appear under the list of
+        loaded modules.
+      \item On a Linux operating system the GNU Project Debugger \code{gdb} can be used
+        for attaching to a running Python process, reading symbol files, and setting breakpoints.
+        The following example attaches to a running process and sets a breakpoint inside
+        an \code{itk.Image} Python object.
+
+\begin{minted}[baselinestretch=1,fontsize=\footnotesize,linenos=false,bgcolor=ltgray]{bash}
+(venv-itk) > sudo gdb
+(gdb) > attach 99999
+Reading symbols from /path/to/ITK-build/Wrapping/Generators/Python/itk/_ITKPyBasePython.so...
+Reading symbols from /path/to/ITK-build/Wrapping/Generators/Python/itk/_ITKCommonPython.so...
+(gdb) > break /path/to/ITK-build/Wrapping/Modules/ITKCommon/itkImagePython.cpp:<ln>
+Breakpoint 1 at ...:
+file /path/to/ITK-build/Wrapping/Modules/ITKCommon/itkImagePython.cpp, line <ln>.
+(gdb) > break /path/to/ITK-source/Core/Common/include/itkImage.hxx:<ln>
+Breakpoint 2 at ...:
+/path/to/ITK-source/Modules/Core/Common/include/itkImage.hxx:<ln>
+(gdb) > c
+
+... continue in Python session until breakpoint is hit ...
+\end{minted}
+
+    \end{itemize}
+
+\end{enumerate}
+
+With these steps completed the respective debugger can be used to set breakpoints
+and step through C++ source code for a respective ITK Python execution.
+
+
```

Comment:
```suggestion
In Visual Studio, debugging can be enabled right from the start if the Python script is loaded into Visual Studio as part of a "Python Project". Mixed mode debugging needs to be enabled, as per documentation:
\url{https://docs.microsoft.com/en-us/visualstudio/python/debugging-mixed-mode-c-cpp-python-in-visual-studio?view=vs-2019}
Starting a project like this is an order of magnitude slower, as many debug symbols need to be loaded and examined.
```

Reply:
I don't know whether `\url{...}` is the right syntax.

Reply:
I believe `\url` is the right syntax, thanks for contributing!